add-dataset/README.md

```diff
@@ -62,11 +56,13 @@ Adding, harvesting, and storing are all verbs used in Data Together to describe
 
 Data Together helps groups steward data they care about by [...]
 
+* Public Record means [...]
+* Data Together Nodes means [...]
+* Distributed Data Stewardship means [...]
```

Comment:
The definition of these may come up in conversation at the 4S event, but generally we should put time into defining these!